python/ray/serve/_private/replica.py

```diff
@@ -498,6 +502,19 @@ def user_health_check():
             )
             self.metrics_pusher.start()
 
+        # Periodically set the gauge metrics to avoid stale values.
+        run_background_task(self._update_metrics_loop())
+
+    async def _update_metrics_loop(self):
+        while True:
+            request_stats = self._get_handle_request_stats()
```

Comment:
Will this call still succeed when the GCS crashes?

Reply:
(Just digging the code) The call doesn't rely on gcs, as long as worker is alive. is there any concern for it?

Reply:
No I just wanted to double check. Since we're running this in a background coroutine, I didn't want it to fail silently when the GCS crashes.